fetch-utils.js

```diff
@@ -4,27 +4,36 @@ const SUPABASE_KEY =
 const client = supabase.createClient(SUPABASE_URL, SUPABASE_KEY);
 
 export async function getCountries(name, continent) {
-    // > Part D: Add a second argument to `select(` to 
+    // > Part D: Add a second argument to `select(` to
     // return an exact db count of matching records
 
     // > Part A: Implement the client query from countries:
     //   1. select all columns
     //   2. order by country name
     //   3. limit to 100 countries
-    let query = // ?
+
+    let query = client.from('countries').select('*', { count: 'exact' }).limit(100).order('name');
 
     if (name) {
+        query = query.ilike('name', `%${name}%`);
         // > Part C: add query for name
     }
```

Comment:
you'll see a lot of this pattern, taking a response or request, passing it to something that modifies the request, then passes it on the something else